uwblueprint/childrens-aid-society
# Squash all intake model migrations

## Notion ticket link

* [Squash migrations from sprint 1](https://www.notion.so/uwblueprintexecs/Squash-migrations-from-Sprint-1-4b86a1c04f6f44c58c1ec53100e3d8c9)
* [Add reference to child in caregivers model](https://www.notion.so/uwblueprintexecs/Add-reference-to-child-in-caregivers-model-71ec1d78cf4b4d5b9638d6ec7fd5b8a8)
* [Create children model](https://www.notion.so/uwblueprintexecs/Create-children-model-031ad31e425d4b0f9151aee5182c9b42)
* [Create visit locations model](https://www.notion.so/uwblueprintexecs/Create-visit_locations-model-89fec44fb1404e4e8b9c8f63fde32f2d)
* [Create intakes model](https://www.notion.so/uwblueprintexecs/Create-intakes-model-a28001141c9c427d96a17a87faa6e6d1)
* [Create intake<>concerns association tables](https://www.notion.so/uwblueprintexecs/Create-association-tables-for-intake-child-concerns-and-intake-familial-concerns-b5d08078495240fb90491d17de0ab12d)
* [Create short/long term goals models](https://www.notion.so/uwblueprintexecs/Create-short-long-term-goals-models-4c8e0ae6e0454d5db22efe4f414ca4a6)



## Implementation description
* Added in the remaining tables required for creating intakes
* Squashed all migrations into one for easier management



## Steps to test
1. Restart your docker containers: `docker-compose down --volumes && docker-compose up`
2. Exec into the backend container: `docker exec -it cas_py_backend /bin/bash`
3. Upgrade the database: `flask db upgrade` -- verify that it works successfully
4. Downgrade and upgrade the database again: `flask db downgrade && flask db upgrade` -- verify that it works successfully
5. Exec into the db container: `docker exec -it cas_db psql -U postgres -d cas`
6. Verify that you see the following tables (there should be 16 + the `alembic_version` table):
```
                   List of relations
 Schema |           Name            | Type  |  Owner
--------+---------------------------+-------+----------
 public | access_types              | table | postgres
 public | addresses                 | table | postgres
 public | alembic_version           | table | postgres
 public | branches                  | table | postgres
 public | caregivers                | table | postgres
 public | child_concerns            | table | postgres
 public | children                  | table | postgres
 public | daytime_contacts          | table | postgres
 public | familial_concerns         | table | postgres
 public | intakes                   | table | postgres
 public | intakes_child_concerns    | table | postgres
 public | intakes_familial_concerns | table | postgres
 public | long_term_goals           | table | postgres
 public | short_term_goals          | table | postgres
 public | transportation_methods    | table | postgres
 public | users                     | table | postgres
 public | visit_locations           | table | postgres
```
7. Check that the tests still run successfully: `docker exec -it cas_py_backend /bin/bash -c "pip install -e . && pytest"`


## What should reviewers focus on?
* does the database still look okay?


## Checklist
